gradle-conjure/src/main/java/com/palantir/gradle/conjure/ConjurePlugin.java

```diff
@@ -471,6 +493,77 @@ private static void setupConjurePythonProject(
+            Task compileIrTask,
+            Configuration conjureGeneratorsConfiguration) {
+        // Validating that each subproject has a corresponding generator.
+        // We do this in afterEvaluate to ensure the configuration is populated.
+        project.afterEvaluate(p -> {
+            Map<String, Dependency> generators = conjureGeneratorsConfiguration
+                    .getAllDependencies()
+                    .stream()
+                    .collect(Collectors.toMap(dependency -> {
+                        Preconditions.checkState(dependency.getName().startsWith(CONJURE_GENERATOR_DEP_PREFIX),
+                                "Generators should start with '%s' according to conjure RFC 002, "
+                                        + "but found name: '%s' (%s)",
+                                CONJURE_GENERATOR_DEP_PREFIX, dependency.getName(), dependency);
+                        return dependency.getName().substring(CONJURE_GENERATOR_DEP_PREFIX.length());
+                    }, Function.identity()));
+
+            project.getChildProjects().entrySet().stream()
+                    .forEach(entry -> {
+                        String subprojectName = entry.getKey();
+                        Project subproject = entry.getValue();
+                        String conjureLanguage = extractSubprojectLanguage(project.getName(), subprojectName);
+                        if (!FIRST_CLASS_GENERATOR_PROJECT_NAMES.contains(conjureLanguage)
+                                && !generators.containsKey(conjureLanguage)) {
+                            throw new RuntimeException(String.format("Discovered subproject %s without corresponding "
+                                            + "generator dependency with name '%s'",
+                                    subproject.getPath(), ConjurePlugin.CONJURE_GENERATOR_DEP_PREFIX + subprojectName));
+                        }
+                    });
+        });
+
+        project.getChildProjects().entrySet().stream()
+                .filter(e -> FIRST_CLASS_GENERATOR_PROJECT_NAMES.stream()
+                        .noneMatch(l -> l.equals(e.getKey().substring(project.getName().length() + 1))))
+                .forEach(e -> {
+                    String subprojectName = e.getKey();
+                    Project subproject = e.getValue();
+                    String conjureLanguage = extractSubprojectLanguage(project.getName(), subprojectName);
+
+                    // We create a lazy filtered FileCollection to avoid using afterEvaluate.
+                    FileCollection matchingGeneratorDeps = conjureGeneratorsConfiguration.fileCollection(
+                            dep -> dep.getName().equals(CONJURE_GENERATOR_DEP_PREFIX + conjureLanguage));
+
+                    ExtractExecutableTask extractConjureGeneratorTask = ExtractExecutableTask.createExtractTask(
+                            project,
+                            GUtil.toLowerCamelCase("extractConjure " + conjureLanguage),
```

Comment:
woah is that space intentional?

Reply:
yes, its to specify the word boundary so that the resulting string is capitalized correctly